conjure-java-core/src/test/java/com/palantir/conjure/java/types/ExternalImportSafetyTests.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.types;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.logsafe.DoNotLog;
+import com.palantir.logsafe.Safe;
+import com.palantir.product.ExternalLongUnionExample;
+import com.palantir.product.SafeExternalLongAlias;
+import com.palantir.product.SafeExternalLongExample;
+import com.palantir.product.UndertowExternalLongTestService;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedParameterizedType;
+import java.lang.reflect.AnnotatedType;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+
+public class ExternalImportSafetyTests {
+
+    @Test
+    public void testAliasAnnotations() {
+        assertThat(SafeExternalLongAlias.class).hasAnnotation(Safe.class);
+
+        assertMethodHasAnnotation(SafeExternalLongAlias.class, "toString", Safe.class);
+        assertMethodHasAnnotation(SafeExternalLongAlias.class, "get", Safe.class);
+        assertMethodParamHasAnnotation(SafeExternalLongAlias.class, "valueOf", "value", Safe.class);
+        assertMethodParamHasAnnotation(SafeExternalLongAlias.class, "of", "value", Safe.class);
+    }
```

Comment:
I could go either way on reflection-based testing. We have tests which generate `integrationInput`, so future changes will have clear diffs which can be reviewed more clearly than reading reflection code, but that does rely on the reviewer to understand and account for all changes.

Reply:
Do you think there's any harm in having both? I see your point about the integrationInput being more readable, but I feel like external import safety is kind of brittle, because if someone forgets to use the SafetyUtils wrappers then we lose safety markings. So I thought a test at least provides _some_ in-your-face protection against future breaks

Reply:
Totally fine to have both :-)